zebrad/src/components/mempool.rs

```diff
@@ -307,7 +324,7 @@ impl Service<Request> for Mempool {
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        let is_state_changed = self.update_state();
+        let is_state_changed = self.update_state(None);
```

Comment:
Moving the `tip_action` fetch to the start of this method might help avoid inconsistencies between the `last_seen_tip_hash` and the actual tip hash the mempool is based on.

```suggestion
        let tip_action = self.chain_tip_change.last_tip_change();
        let is_state_changed = self.update_state(tip_action);
```